File: app/src/main/java/com/microlabs/trallet/adapter/CurrencySpinnerAdapter.java
package com.microlabs.trallet.adapter;

import android.content.Context;
import android.support.annotation.LayoutRes;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.TextView;

import com.microlabs.trallet.R;
import com.microlabs.trallet.base.RealmHelper;
import com.microlabs.trallet.model.Category;
import com.microlabs.trallet.model.Currency;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by Nicholas on 3/13/2017.
 */

public class CurrencySpinnerAdapter extends ArrayAdapter<Currency> {

    private ArrayList<Currency> mCurrency;
    private LayoutInflater inflater;

    public CurrencySpinnerAdapter(Context context, int resource, ArrayList<Currency>list) {
        super(context, resource, list);
        mCurrency = list;
        inflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
    }

    @Override
    public View getDropDownView(int position, @Nullable View convertView, @NonNull ViewGroup parent) {
        View v = inflater.inflate(R.layout.item_category,parent,false);
        TextView lblCategory = (TextView) v.findViewById(R.id.lblCategoryText);
        lblCategory.setText(mCurrency.get(position).getName());
        return v;
    }

    @NonNull
    @Override
    public View getView(int position, @Nullable View convertView, @NonNull ViewGroup parent) {
        View v = inflater.inflate(R.layout.item_category,parent,false);
        TextView lblCategory = (TextView) v.findViewById(R.id.lblCategoryText);
        lblCategory.setText(mCurrency.get(position).getName());
        return v;
    }

    public int getPosition(int currId) {
        return super.getPosition(RealmHelper.getCurrencyById(currId));
    }
}

File: app/src/main/java/com/microlabs/trallet/adapter/CurrencySpinnerAdapter.kt
package com.microlabs.trallet.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import com.microlabs.trallet.R
import com.microlabs.trallet.model.Currency
import kotlinx.android.synthetic.main.item_category.view.*

/**
 * Created by Nicholas on 3/13/2017.
 *
 * RCurrency Spinner Custom Adapter
 */

class CurrencySpinnerAdapter(context: Context, resource: Int, private val currencyList: List<Currency>) : ArrayAdapter<Currency>(context, resource, currencyList) {
    private val inflater: LayoutInflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
    private var itemView: View? = null

    override fun getDropDownView(position: Int, convertView: View?, parent: ViewGroup): View {
        val v = inflater.inflate(R.layout.item_category, parent, false)
        v.lblCategoryText.text = currencyList[position].name
        return v
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        if (itemView == null) {
            itemView = inflater.inflate(R.layout.item_category, parent, false)
        }
        itemView!!.lblCategoryText.text = currencyList[position].name
        return itemView!!
    }

    fun getPositionById(currencyId: Int): Int {
        return currencyList.indexOfFirst { it.id == currencyId }
    }
}
